core/logger/logger.go

```diff
@@ -123,7 +134,7 @@ func newProductionConfig(dir string, jsonConsole bool, toDisk bool, unixTS bool)
 }
 
 type Config interface {
-	RootDir() string
+	RootDir(Logger) string
```

Comment:
This is a bit of an oddity - if we leave this injection pattern, perhaps this case should return the error.